src/client/routes/Portfolio/index.tsx

```diff
@@ -24,124 +28,154 @@ const HeaderCard = styled(SummaryCard)`
-  const currentNetworkSettings = NETWORK_SETTINGS[currentNetwork];
-  const vaultsSummary = useAppSelector(VaultsSelectors.selectSummaryData);
   // const labsSummary = useAppSelector(LabsSelectors.selectSummaryData);
-  const walletSummary = useAppSelector(TokensSelectors.selectSummaryData);
   const userTokens = useAppSelector(TokensSelectors.selectUserTokens);
   const activeModal = useAppSelector(ModalSelectors.selectActiveModal);
   const appStatus = useAppSelector(AppSelectors.selectAppStatus);
   const tokensListStatus = useAppSelector(TokensSelectors.selectWalletTokensStatus);
   const generalLoading = (appStatus.loading || tokensListStatus.loading || isMounting) && !activeModal;
-  const userTokensLoading = generalLoading && !userTokens.length;
+  const userAddress: string | undefined = location.pathname.split('/')[2];
+  const userPortfolio = useAppSelector(LinesSelectors.selectUserPortfolio);
 
-  const netWorth = toBN(vaultsSummary.totalDeposits)
-    .plus(walletSummary.totalBalance)
-    // .plus(labsSummary.totalDeposits)
-    .toString();
-
-  const summaryCardItems = [
-    { header: t('dashboard.total-net-worth'), Component: <Amount value={netWorth} input="usdc" /> },
-  ];
-  if (walletIsConnected) {
-    summaryCardItems.push({
-      header: t('dashboard.available-deposit'),
-      Component: <Amount value={walletSummary.totalBalance} input="usdc" />,
-    });
-  }
-  if (currentNetworkSettings.earningsEnabled) {
-    summaryCardItems.push(
-      {
-        header: t('dashboard.vaults-earnings'),
-        Component: <Amount value={vaultsSummary.totalEarnings} input="usdc" />,
-      },
-      {
-        header: t('dashboard.vaults-est-yearly-yield'),
-        Component: <Amount value={vaultsSummary.estYearlyYeild} input="usdc" />,
+  const userTokensLoading = generalLoading && !userTokens.length;
+  const [currentRole, setRole] = useState<string>(BORROWER_POSITION_ROLE);
+  const [data, setdata] = useState<any[]>([]);
+  const [aggregatedCreditLinePage, setAggregatedCreditLine] = useState<CreditLinePage>();
+
+  const availableRoles = [BORROWER_POSITION_ROLE, LENDER_POSITION_ROLE];
+
+  const SummaryCardItems = availableRoles.map((role: string) => {
+    return {
+      header: t(''),
+      Component: (
+        <RoleOption onClick={() => handleSetRole(role)} active={role === currentRole} key={`s-${role}`}>
+          {t(`settings:${role}`)}
+        </RoleOption>
+      ),
+    };
+  });
+
+  const handleSetRole = (role: string) => {
+    setdata([]);
+    setRole(role);
+  };
+
+  useEffect(() => {
+    if (!userAddress || !isValidAddress(userAddress)) {
+      dispatch(AlertsActions.openAlert({ message: 'INVALID_ADDRESS', type: 'error' }));
+      return;
+    } else if (userAddress.length === 42) {
+      dispatch(LinesActions.getUserPortfolio({ user: userAddress.toLocaleLowerCase() }));
+    }
+  }, [currentRole, walletIsConnected]);
+
+  useEffect(() => {
```

Comment:
shouldnt need a useEffect for just switching what data we already have is presented.
Also why do we need to call `setSelectedLineAddress` if the page covers multiples lines?

Reply:
@Seroxdesign Are you taking this?